source/AMDSClient.h

```diff
@@ -31,6 +32,7 @@ private slots:
 	void onSocketDataReady(AMDSClientTCPSocket* socket, AMDSClientRequest *clientRequest);
 
 	void onSocketError(AMDSClientTCPSocket *clientTCPSocket, QAbstractSocket::SocketError socketError);
+	void onRequestTypeChanged(QString);
```

Comment:
Especially for QObjects, it's better to make sure you pass non-pointer values with a `const QString &` paradigm.  Then it uses the implicit sharing from Qt and is faster (not that apparent when using QString, but important when passing large QVector's or some such around).